frontends/ast/simplify.cc

```diff
@@ -2886,10 +2886,15 @@ AstNode *AstNode::readmem(bool is_readmemh, std::string mem_filename, AstNode *m
 	int meminit_size=0;
 
 	std::ifstream f;
+	std::string path = filename.substr(0, filename.find_last_of("\\/")+1);
 	f.open(mem_filename.c_str());
-	yosys_input_files.insert(mem_filename);
-
-	if (f.fail())
+	if (f.fail()) {
+		f.open(path + mem_filename.c_str());
+		yosys_input_files.insert(path + mem_filename);
+	} else {
+		yosys_input_files.insert(mem_filename);
+	}
+	if (f.fail() || strlen(mem_filename.c_str()) == 0)
```

Comment:
I added `strlen(mem_filename.c_str()` to avoid a hung when the included file is "" and `path != ""`. As I said, I am not a C++ expert, so maybe there is an option to fail when not a file? (I didn't find in a quick search).

Reply:
Please use `GetSize(mem_filename) == 0` rather than `strlen(mem_filename.c_str()) == 0`

Reply:
Done. I saw that it is specified in the CodingReadme, sorry.